hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/trigger/TriggerAction.java

```diff
@@ -0,0 +1,386 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.alerts.api.model.trigger;
+
+import java.io.Serializable;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+
+/**
+ * Link an ActionDefinition with a Trigger.
+ *
+ * It can add optional constraints that determine when an action will be executed:
+ *
+ * - A set of Alert.Status (represented by its string value).
+ *   The action will be executed if the Alert which is linked is on one of the states defined.
+ *   This is not applicable if the action is linked with an Event.
+ *
+ * - A calendar expression that defines an interval when the action will be executed.
+ *   The format of the calendar can be:
+ *   - Absolute: <startAbsoluteDate>;<endAbsoluteDate>
+ *       With <startAbsoluteDate> and <endAbsoluteDate> format as yyyy-MM-dd.HH:mm
+ *   - Relative: <startRelativeDate>;<endRelativeDate>
+ *       With <startRelativeDate> and <endRelativeDate> format as
+ *       R[[M<Month_of_the_year>.]D<Day_of_the_week>.]HH:mm
+ *
+ *       <Month_of_the_year> uses Calendar.MONTH numeration
+ *       <Day_of_the_week> uses Calendar.DAY_OF_WEEK numeration
+ *
+ * @author Jay Shaughnessy
+ * @author Lucas Ponce
+ */
+public class TriggerAction implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+    private static final SimpleDateFormat parser = new SimpleDateFormat("yyyy-MM-dd.HH:mm");
+
+    public enum CalendarRelative {
+        MONTH, DAY_OF_THE_WEEK, HOUR_OF_THE_DAY, MINUTE
+    }
+
+    @JsonInclude(Include.NON_NULL)
+    private String tenantId;
+
+    @JsonInclude
+    private String actionPlugin;
+
+    @JsonInclude
+    private String actionId;
+
+    @JsonInclude(Include.NON_EMPTY)
+    Set<String> states;
+
+    @JsonInclude(Include.NON_NULL)
+    private String calendar;
```

Comment:
Two comments:
- Related to use a calendar expression versus a TimeConstraint object: I've found a expression more simple to read. As this object has a strong mapping with json. We can add some static helpers at that level to do more easy to create it, but for readability I would like to avoid complex nested objects on the   json representation.

- About where to place the constraint, yes, there are several candidate places. My goal for this design was to maintain previous behaviour as simple as we had, but also introduce this. I was thought which place could be more interesting, to place a time constraint on the link with the trigger or in the action itself. Well, both are valid, but I think that more common scenario will be to use in the trigger. For example, an action definition will be used more like a "group" (i.e. I may have an action to notify sysadmins, another action to notify developers, and another action for managers). So, as action definitions can be reused over multiple triggers, I thought that the constraints should live in the relation between trigger and action, hence that I put the logic there.

Of course, we can add both, in the link and the action definition, and add more complexity in the rules to check, but I would like to add that in an iterative way to check how these features looks like in real scenarios.